overlord/snapstate/snapstate.go

```diff
@@ -1969,17 +1969,129 @@ func updateManyFiltered(ctx context.Context, st *state.State, names []string, re
 		return nil, nil, err
 	}
 
-	updated, updateTss, err := doUpdate(ctx, st, names, toUpdate, params, userID, flags, nil, deviceCtx, fromChange)
-	if err != nil {
-		return nil, nil, err
+	// if we're on classic with a kernel/gadget, split refreshes with essential
+	// snaps and apps so that the apps don't have to wait for a reboot
+	essential, nonEssential := splitEssentialUpdates(deviceCtx, toUpdate, names)
```

Comment:
Looking at were returned values are used, this should go inside the "if" just below, that would also match the comment here.

Reply:
We need this to check that there is a mix of snaps that makes the split refresh possible. I added a helper in my previous refactor that makes this a bit nicer